khmer/_oxli/graphs.pyx

```diff
@@ -211,87 +212,78 @@ cdef class Hashtable:
                                                                    max_count))
         return posns
 
-    def consume_seqfile_with_reads_parser(self, read_parser):
-        """Count all k-mers from read_parser."""
-        cdef unsigned long long n_consumed = 0
-        cdef unsigned int total_reads = 0
-
-        cdef CPyReadParser_Object* parser = <CPyReadParser_Object*>read_parser
+    cdef FastxParserPtr _get_parser(self, object parser_or_filename) except *:
+        cdef FastxParserPtr _parser
+        if isinstance(parser_or_filename, FastxParser):
+            _parser = (<FastxParser>parser_or_filename)._this
+        elif isinstance(parser_or_filename, ReadParser):
+            _parser = (<CPyReadParser_Object*>parser_or_filename).parser
+        else:
```

Comment:
I was thinking -- there's a function in `utils` called `is_str`, which just checks for, well, `str`ness. It might be better to do another `elif` with that here, and then raise a `TypeError` on the `else`. Otherwise, any `TypeError` is going to be raised by `_bstring`, which will be confusing to the user.

Reply:
Good thought. Updated!